### PR TITLE
Corrected code syntax

### PR DESCRIPTION
**Pull request by RinRamz**
Cleaned code

### CODE REVIEW

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/CamaraScript.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using UnityEngine;
+
+namespace Ninja
+{
+    public class CamaraScript : MonoBehaviour
+    {
+        [SerializeField] private GameObject player = default;
+        void Update()
```
[the remark this answers is not in the document]
Comment (Horothenic) [reply Ninja/Assets/_Main/PlayerAssets/Player's Scripts/CamaraScript.cs:8]: Underscore missing.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerAttackingState.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            if (_context.IsAttackPressed && !_context.IsInFirstAttack)
+            {
+                _context.Animator.Play("Player_Attack2");
+                _context.IsInSecondAttack = true;
+            }
+        }
+        public override void ExitState()
+        {
+        }
+        public override void CheckIfSwitchStates()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerAttackingState.cs:28]: Add spaces between methods.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerLandingState.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        public override void UpdatetState()
+        {
+            CheckIfSwitchStates();
+        }
+        public override void ExitState()
+        {
+        }
+        public override void CheckIfSwitchStates()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerLandingState.cs:21]: Add space between methods.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerMovingState.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        public override void UpdatetState()
+        {
+            CheckIfSwitchStates();
+            _context.Flip();
+            _context.Move();
+        }
+        public override void ExitState()
+        {
+        }
+        public override void CheckIfSwitchStates()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerMovingState.cs:24]: Add space between methods.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerStateFactory.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        public PlayerStateFactory(PlayerStateManager currentContext)
+        {
+            _context = currentContext;
+        }
+
+        public PlayerBaseState Idle()
+        {
+            return new PlayerIdleState(_context, this);
+        }
+        public PlayerBaseState Jump()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerStateFactory.cs:16]: Add space between methods.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerStateManager.cs [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        {
+            _dust = GetComponentInChildren<ParticleSystem>();
+            _animator = GetComponent<Animator>();
+            _rigidbody2D = GetComponent<Rigidbody2D>();
+
+            _states = new PlayerStateFactory(this);
+            _currentState = _states.Idle();
+            _currentState.EnterState();
+        }
+        private void Update()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerStateManager.cs:70]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyBasicStateMachine/EnemyBasicBaseState.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    public abstract class EnemyBasicBaseState
+    {
+        protected EnemyBasicStateManager _context;
+        protected EnemyBasicStateFactory _enemyBasicStateFactory;
+        public EnemyBasicBaseState(EnemyBasicStateManager currentContext, EnemyBasicStateFactory enemyBasicStateFactory)
+        {
+            _context = currentContext;
+            _enemyBasicStateFactory = enemyBasicStateFactory;
+        }
+        public abstract void EnterState();
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyBasicStateMachine/EnemyBasicBaseState.cs:12]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyBasicStateMachine/EnemyBasicStateManager.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        private float _sightRange = 6f;
+        private float _attackRange = .6f;
+        private float _movementSpeed = 4.5f;
+        private float _attackSpeed = 1f;
+        private float _nextAttackTime = 0f;
+        private bool _inRangeofSight = false;
+        private bool _inRangeofAttack = false;
+        private bool _receivedDamage = default;
+        private bool _isCrit = default;
+
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyBasicStateMachine/EnemyBasicStateManager.cs:28]: Remove extra space.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyBasicStateMachine/EnemyBasicStateManager.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                _receivedDamage = false;
+                _isCrit = false;
+            }
+            else if (collision.CompareTag("Attack2Collider"))
+            {
+                _receivedDamage = false;
+                _isCrit = false;
+            }
+
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyBasicStateMachine/EnemyBasicStateManager.cs:93]: Remove extra space.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterAttackingState.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+            _context.Animator.Play("Player_Iddle");
+            _context.NextAttackTime = _context.EnemyActions.AttackRanged(_context.NextAttackTime, _context.AttackSpeed, _context.Bullet);
+        }
+
+        public override void UpdatetState()
+        {
+            CheckIfSwitchStates();
+            _context.NextAttackTime = _context.EnemyActions.AttackRanged(_context.NextAttackTime, _context.AttackSpeed, _context.Bullet);
+        }
+        public override void ExitState()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterAttackingState.cs:18]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterBaseState.cs [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    public abstract class EnemyShooterBaseState
+    {
+        protected EnemyShootherStateManager _context;
+        protected EnemyShooterStateFactory _enemyShooterStateFactory;
+        public EnemyShooterBaseState(EnemyShootherStateManager currentContext, EnemyShooterStateFactory enemyShooterStateFactory)
+        {
+            _context = currentContext;
+            _enemyShooterStateFactory = enemyShooterStateFactory;
+        }
+        public abstract void EnterState();
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterBaseState.cs:12]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterIdleState.cs [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        public override void EnterState()
+        {
+            _context.Animator.Play("Player_Iddle");
+        }
+
+        public override void UpdatetState()
+        {
+            CheckIfSwitchStates();
+        }
+        public override void ExitState()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterIdleState.cs:17]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterMovingState.cs [hunk 13]
```diff
@@ -0,0 +1,36 @@
+namespace Ninja
+{
+    public class EnemyShooterMovingState : EnemyShooterBaseState
+    {
+        public EnemyShooterMovingState(EnemyShootherStateManager currentContext, EnemyShooterStateFactory enemyShooterStateFactory)
+        : base(currentContext, enemyShooterStateFactory) { }
+
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterMovingState.cs:7]: Remove extra space

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterPatrollingState.cs [hunk 14]
```diff
@@ -0,0 +1,23 @@
+namespace Ninja
+{
+    public class EnemyShooterPatrollingState : EnemyShooterBaseState
+    {
+        public EnemyShooterPatrollingState(EnemyShootherStateManager currentContext, EnemyShooterStateFactory enemyShooterStateFactory)
+        : base(currentContext, enemyShooterStateFactory) { }
+        public override void EnterState()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShooterPatrollingState.cs:7]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShootherStateManager.cs [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+        public Transform PlayerPos => _player;
+        public Rigidbody2D Rigidbody2D => _rigidBody2D;
+        public GameObject Bullet => _bullet;
+        public Animator Animator => _animator;
+        public float NextAttackTime { get { return _nextAttackTime; } set { _nextAttackTime = value; } }
+        public bool InRangeOfSight { get { return _inRangeofSight; } set { _inRangeofSight = value; } }
+        public bool InRangeOfAttack { get { return _inRangeofAttack; } set { _inRangeofSight = value; } }
+        public float AttackSpeed => _attackSpeed;
+        public float MovementSpeed => _movementSpeed;
+        private void Awake()
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShootherStateManager.cs:40]: Add space between methods.

File: Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShootherStateManager.cs [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+                _hp = _enemyActions.GetDamaged(damage, _hp);
+            }
+            else if (collision.CompareTag("Attack2Collider"))
+            {
+                int damage = collision.GetComponentInParent<PlayerStateManager>().Damage;
+                _hp = _enemyActions.GetDamaged(Mathf.RoundToInt(damage * 1.75f), _hp);
+            }
+        }
+
+
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/EnemyScripts/EnemyShooterStates/EnemyShootherStateManager.cs:73]: Remove extra spaces.

File: Ninja/Assets/_Main/Scripts/Singleton.cs [hunk 17]
```diff
@@ -0,0 +1,33 @@
+using UnityEngine;
+
+namespace Ninja
+{
+    public class Singleton<T> : MonoBehaviour where T : Component
+    {
+        private static T _instance;
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/Singleton.cs:7]: Initizalize.

File: Ninja/Assets/_Main/Scripts/Singleton.cs [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    GameObject gameObject = new GameObject();
+                    gameObject.name = typeof(T).Name;
+                    gameObject.hideFlags = HideFlags.HideAndDontSave;
+                    _instance = gameObject.AddComponent<T>();
+
```
Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/Singleton.cs:19]: Remove extra space.

Comment (Horothenic) [inline Ninja/Assets/_Main/Scripts/Singleton.cs:19]: Remove extra space.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/CamaraScript.cs [hunk 19]
```diff
@@ -0,0 +1,15 @@
+using UnityEngine;
+
+namespace Ninja
+{
+    public class CamaraScript : MonoBehaviour
+    {
+        [SerializeField] private GameObject player = default;
```
Comment (Prot089) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/CamaraScript.cs:7]: Undersocre missing

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerAttackingState.cs (hunk 2 above)
Comment (Prot089) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerAttackingState.cs:28]: Add spaces between methods.

File: Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerBaseState.cs [hunk 20]
```diff
@@ -0,0 +1,24 @@
+namespace Ninja
+{
+    public abstract class PlayerBaseState
+    {
+        protected PlayerStateManager _context;
```
Comment (Prot089) [inline Ninja/Assets/_Main/PlayerAssets/Player's Scripts/PlayerStates/PlayerBaseState.cs:5]: Initialize "_context"

File: Ninja/Assets/_Main/Scripts/Singleton.cs (hunk 17 above)
Comment (Prot089) [inline Ninja/Assets/_Main/Scripts/Singleton.cs:7]: Initizalize "_instance"

**Review by Prot089 — COMMENTED** [review]
Just remember to use the auto format :)

**Review by Horothenic — APPROVED** [review]

**Pull request merged**